velox/exec/tests/AggregationTest.cpp

```diff
@@ -3038,4 +3040,143 @@ TEST_F(AggregationTest, memoryPoolAllocationAtInit) {
 
   assertQuery(plan, "SELECT c0, cast(NULL as INT) FROM tmp GROUP BY c0");
 }
+
+DEBUG_ONLY_TEST_F(AggregationTest, reclaimEmptyInput) {
+  constexpr int64_t kMaxBytes = 1LL << 30; // 1GB
+  auto rowType = ROW({"c0", "c1", "c2"}, {INTEGER(), INTEGER(), VARCHAR()});
+  const int32_t numBatches = 5;
+  std::vector<RowVectorPtr> batches;
+  VectorFuzzer fuzzer({.vectorSize = 100}, pool());
+  for (int32_t i = 0; i < numBatches; ++i) {
+    batches.push_back(fuzzer.fuzzInputRow(rowType));
+  }
+
+  std::shared_ptr<Task> task;
+  std::atomic<bool> injectReclaimOnce{true};
+  SCOPED_TESTVALUE_SET(
+      "facebook::velox::exec::Values::getOutput",
+      std::function<void(const exec::Values*)>([&](const exec::Values* values) {
+        if (!injectReclaimOnce.exchange(false)) {
+          return;
+        }
+        auto* driver = values->testingOperatorCtx()->driver();
+        task = values->testingOperatorCtx()->task();
+        {
+          MemoryReclaimer::Stats stats;
+          SuspendedSection suspendedSection(driver);
+          task->pool()->reclaim(kMaxBytes, stats);
```

Comment:
Yes, the get output test injection here is to trigger memory arbitration at the start of the query execution.

I had two theories that the empty spilling check failure is caused by empty input, and the memory arbitrator tries to spill from the empty aggregation operator. It is not the case and grouping set has the empty check before it triggers spilling. I leave the test case here to in case any future code change.